test/unit/lookup_key_test.rb

```diff
@@ -24,7 +24,7 @@ def test_path2match_single_domain_path
     key   = ""
     value = ""
     as_admin do
-      key   = LookupKey.create!(:key => "ntp", :path => "domain", :puppetclass => Puppetclass.first)
+      key   = LookupKey.create!(:key => "ntp", :path => "domain", :puppetclass => Puppetclass.first, :key_type => "string")
```

Comment:
For invalid values of key_type, sure, throw an error.  Blank/nil is a valid value though (as is string, etc), so it shouldn't throw an error.  Probably simply skip casting/validation `unless key_type.present?`.

The current code quietly ignores invalid key_types, but that was the earlier point - it's perfectly safe to raise an error there.